src/main/java/seedu/address/model/person/Person.java

```diff
@@ -102,13 +103,27 @@ public Address getAddress() {
         return address.get();
     }
 
+    public void setRemark(Remark remark) {
+        this.remark.set(requireNonNull(remark));
+    }
+
+    @Override
+    public ObjectProperty<Remark> remarkProperty() {
+        return remark;
+    }
+
+    @Override
+    public Remark getRemark() {
+        return remark.get();
+    }
+
     /**
      * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
      * if modification is attempted.
```

Comment:
`Collections.unmodifiableSet()` is used as a form of defensive programming to ensure that your getter method cannot have setter method behaviors. However, as `UniqueTagList#toSet()` returns a set that is change-insulated against the internal list, it is not a problem to remove this `Collections.unmodifiableSet()`.

However, removing `Collections.unmodifiableSet()` means that you need to accordingly update the header comment of this method.